src/cloudai/systems/slurm/slurm_system.py

```diff
@@ -178,17 +178,12 @@ def supports_gpu_directives(self) -> bool:
             self.supports_gpu_directives_cache = True
             return True
 
-        has_gres_gpu = False
-        has_gpu_gres_type = False
-
         for line in stdout.splitlines():
-            if "AccountingStorageTRES" in line and "gres/gpu" in line:
-                has_gres_gpu = True
-            if "GresTypes" in line and "gpu" in line and "(null)" not in line:
-                has_gpu_gres_type = True
+            if "GresTypes" in line and "gpu" in line:
+                self.supports_gpu_directives_cache = True
+                return True
 
-        self.supports_gpu_directives_cache = has_gres_gpu and has_gpu_gres_type
```

Comment:
We still need caching `False` value. Worse adding unit test to check if cache is working for both values.